controllers/apps/transformer_cluster_service.go

```diff
@@ -167,22 +177,68 @@ func (t *clusterServiceTransformer) buildService(transCtx *clusterTransformConte
 		Optimize4ExternalTraffic()
 
 	if len(clusterService.ComponentSelector) > 0 {
+		builder.AddSelector(constant.KBAppComponentLabelKey, clusterService.ComponentSelector)
+	}
+
+	if len(clusterService.ShardingSelector) > 0 {
```

Comment:
component and sharding selectors should be mutually exclusive

Reply:
> component and sharding selectors should be mutually exclusive

It has been documented in the API and runtime validation has been added.